crates/router/src/connector/authorizedotnet/transformers.rs

```diff
@@ -53,24 +61,42 @@ struct CreditCardDetails {
     card_code: Option<masking::Secret<String>>,
 }
 
-#[derive(Serialize, Deserialize, PartialEq, Debug)]
+#[derive(Serialize, Deserialize, Debug)]
 #[serde(rename_all = "camelCase")]
 struct BankAccountDetails {
     account_number: masking::Secret<String>,
 }
 
-#[derive(Serialize, Deserialize, PartialEq, Debug)]
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
 enum PaymentDetails {
-    #[serde(rename = "creditCard")]
     CreditCard(CreditCardDetails),
-    #[serde(rename = "bankAccount")]
     BankAccount(BankAccountDetails),
-    Wallet,
-    Klarna,
-    Paypal,
-    #[serde(rename = "bankRedirect")]
+    OpaqueData(WalletDetails),
+    PayPal(PayPalDetails),
     BankRedirect,
-    BankTransfer,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct PayPalDetails {
+    pub success_url: Option<String>,
+    pub cancel_url: Option<String>,
+}
+
+#[derive(Serialize, Debug, Deserialize)]
```

Comment:
need both Serialize and Deserialize

Reply:
Parent struct uses both deserialize and serialize .